cmd/iam-storj-store.go

```diff
@@ -87,6 +87,7 @@ func (iamOS *IAMStorjAuthStore) loadUser(ctx context.Context, user string, userT
 		return err
 	}
 	req.Header.Set("Authorization", "Bearer "+iamOS.authToken)
+	req.Header.Set("Forwarded", "for="+logger.GetReqInfo(ctx).RemoteHost)
```

Comment:
I assume we are implicitly ignoring existing `Forwaded` or `X-Forwarded-For` to avoid passing malicious clients (?).

Reply:
It's probably obvious; I will approve. :)

Reply:
We probably do need to know what we get from Traefik.

Reply:
It looks that Traefik forward the client IP using x-forwarded-for header https://github.com/traefik/traefik/issues/2311

It even allows trusting in some IPs for forwarding the received forwarded headers https://doc.traefik.io/traefik/routing/entrypoints/#forwarded-headers

Currently, we run Traefik as the entry point to Linksharing and Gateway-MT, at least in the DC and Equinix (https://storj.slack.com/archives/CSXU0Q8P3/p1625589360477900)

Can you confirm if `logger.GetReqInfo(ctx).RemoteHost` is the value received from some heads? otherwise, I'll have a look to confirm it before merging this PR

Reply:
I've checked the source and it looks that's pulling the client IP from the headers and falling back to the request IP if there isn't any header with such information (see the commit message for more details, it has been updated), so it should work fine with our current setup that runs the Gateway-MT in front of a gateway/load balancer (i.e. Traefik).

Gateway-MT will allow to use the headers or not for determining the client IP as whitelisting IPs that are trusted to use the headers but it will only be available once we deploy the in progress refactoring and for now, it's fine to always use the headers because the production setup won't change from running in from of a trusted gateway/load balancer to run it bare.

Reply:
I checked the `GetSourceIPFromHeaders` source, and it does what you mentioned (It's worth pointing out that MinIO is going through all of them as a fallback: X-Forwarder-For, X-Real-IP and Forwarder. Remote address is at the end). Are you suggesting to skip making pulling info from headers optional for now since we will be behind a trusted load balancer for quite some time anyway? I think I'm okay with this, but I would like to get Bill's thoughts on this.

Reply:
No, I'm going to make them optional but it won't work for the Gateway-MT because on what Bill commented in the first comment of this conversation https://review.dev.storj.io/c/storj/gateway-mt/+/5182/1/pkg/server/signature.go#56

Because we should start to use soon the part of the code, in the Gateway-MT, that does the request to the Auth Service, rather than this one, I'm not going to make them optional in this fork because it doesn't look that the effort would pay off.

Reply:
Ack

Reply:
Ack